### PR TITLE
Am pantry class

### PR DESCRIPTION
**Pull request by aemcdonald**
What’s this PR do?
Created the Pantry class
Created initial tests for pantry class
Added an s to ingredientsData on line 300 in scripts.js to fix a variable typo
Created a method in Pantry.js that shows the ingredients in a user's pantry 
Created a method in Pantry.js that determines the ingredients a user has to cook a given recipe
Created a method in Pantry.js that determines the ingredient's missing from a user's pantry to cook a given recipe

Where should the reviewer start?  
line 1 of Pantry.js
line1 of Pantry-test.js

How should this be manually tested?  
Run npm test

Any background context you want to provide?  
Will need to refactor getRecipeIngredientsInStock & getMissingIngredients to determine based on amount
Pantry class is still incomplete, saving work for today

What are the relevant tickets?  
closes issue #17 
closes issue #18 
closes issue #54 

Screenshots (if appropriate)  

Questions:

### CODE REVIEW

File: src/Pantry.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return item.id === ingredient.id;
+      })
+      if (!ingredientNotFound) {
+        missingIngred.push(ingredient)
+      }
+      return missingIngred
+    }, [])
+    console.log(missingIngredients)
+    return missingIngredients
+  }
```
Comment (npdarrington) [inline src/Pantry.js:34]: Hanna and I talked about this on a call already. Now that we can see how both functions are working separately, we are pretty confident that we can compress the getMissingIngredients function and incorporate it (maybe as a helper function?) inside of getRecipeIngredientsInStock. I know that was the original intention, it was a smart decision to build another method out separately to see how it worked and come back to refactor it after seeing them both side by side.

File: test/Pantry-test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    expect(Pantry).to.be.a('function');
+  });
+
+  it('should show the ingredients in a user\'s pantry', () => {
+  expect(pantry.getPantryItems(user)).to.deep.equal(user.pantry);
+  });
+
+  it('should show a user missing ingredients to cook a meal', () => {
+    const recipe = recipeData[0].ingredients;
+    const missingIngredients = []
```
Comment (npdarrington) [inline test/Pantry-test.js:28]: This variable can probably be removed since we got it to start returning as intended without using it.

**Review by npdarrington — APPROVED** [review]
Overall, really solid work on getting Pantry started! I know this one started a lot more difficult than expected. Really appreciate the hard work on it! Gave us all a good start and now we can tackle it with the forward momentum.

File: src/Pantry.js (hunk 1 above)
Comment (hannakim91) [reply src/Pantry.js:34]: Added a ticket for this!

**Review by hannakim91 — APPROVED** [review]

**Review by JWoltjen — APPROVED** [review]
Looks great everyone!

**Pull request merged**